### DISCUSSION TITLE
Why does CommandHandler, resolveHandlerForAggregateRoot look at constructor parameters?

### DISCUSSION THREAD

**Discussion by rikzwarthoff** [spatie/laravel-event-sourcing#335] [Ideas]
The CommandHandler, resolveHandlerForAggregateRoot function resolves the $uuidField by looking at which constructor parameter has the AggregateUuid attribute. In my opinion it would make more sense to look at class properties which have the attribute as it also tries to access a class property.

This would allow for some different constructor parameters and maybe some logic in the constructor. For example setting the uuid by a given model in the constructor. It would also allow for extending of class, say want you put the AggregateUuid attribute in the parent class. Now that's not possible.

<img width="777" alt="Screenshot 2022-03-31 at 19 00 42" src="https://user-images.githubusercontent.com/9021840/161110213-77b36240-2baf-47d9-b7d6-e0adc13619da.png">
.

**Comment by freekmurze** [comment]
@brendt Do you want to take a look at this one?

**Comment by brendt** [comment]
This is by design. 

- Commands themselves should be simple data classes, that means they should only contain public (readonly) properties and getters.
- We encourage (and assume most people do) use promoted properties, which is why we're looking at the constructor arguments. 
- You shouldn't extend command classes, that goes against their purpose.

In summary: it's by design to prevent developers from doing too much in their command classes.
